app/main.py

```diff
@@ -1,3 +1,29 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    database = []
+
+    def inner(*args, **kwargs) -> Callable:
+        in_cache = [
+            data["result"] for data in database if data["arguments"] == args
+        ]
+        if in_cache:
+            print("Getting from cache")
+            return in_cache[0]
+        print("Calculating new result")
+        result = func(*args)
+        database.append({"result": result, "arguments": args})
+        return result
+
+    return inner
```

Comment:
Your approach is incorrect: you use a list and every time you need to iterate through it and spend extra time.
Hint: use a dictionary as the storage